packages/langium/src/references/naming.ts

```diff
@@ -12,7 +12,7 @@ export interface NamedAstNode extends AstNode {
 }
 
 export function isNamed(node: AstNode): node is NamedAstNode {
-    return (node as NamedAstNode).name !== undefined;
+    return typeof (node as NamedAstNode).name === 'string';
```

Comment:
Will it still be possible to have a `name` property number typed? 
```
Test : 
   name: TEST_NUMBER
 ;
terminal TEST_NUMBER returns number: /[0-9]+(\.[0-9])?/;
```

Reply:
Just checked some things. You can still type the `name` prop as `number`, that's ok.

As for a number as an identifying cross reference, I tested something out on this PR and main. Seems in either case numbers don't bind as cross refs. I'm not sure if this is also the case on the 0.3 release itself, but that part seems to be consistent.

However, what this does change is interesting. Whereas before numbers values don't show up as an identifier for a symbols in VSCode before, they can with this PR like any other 'name' value.

<img width="128" alt="Screen Shot 2022-05-06 at 14 06 12" src="https://user-images.githubusercontent.com/5070304/167128512-8e5bf817-6219-41be-a1dd-8597890d4fc8.png">

But, when I extended the `isNamed` check to also factor in number types directly, it still works, but it sometimes leads to reads to a render error. In this case, only when where was a preceding statement.
<img width="234" alt="Screen Shot 2022-05-06 at 14 18 34" src="https://user-images.githubusercontent.com/5070304/167130102-ee750901-e0a1-416a-83b3-a37910ca7046.png">

I like the idea of accounting for numbers directly, but I'm not yet sure how to proceed given the render error.

Reply:
If you want numbers as names for cross-references, you should provide a custom NameProvider that converts the numbers to strings.